src/status_im/contexts/keycard/pin/events.cljs

```diff
@@ -0,0 +1,21 @@
+(ns status-im.contexts.keycard.pin.events
+  (:require [utils.re-frame :as rf]))
+
+(rf/reg-event-fx :keycard.pin/delete-pressed
+ (fn [{:keys [db]}]
+   (let [pin (get-in db [:keycard :pin :text])]
+     (when (and pin (pos? (count pin)))
+       {:db (-> db
+                (assoc-in [:keycard :pin :text] (.slice pin 0 -1))
```

Comment:
Any reason why `.slice` is being used instead of a native CLJS function?

Reply:
what native CLJS function?

Reply:
`subs` is the equivalent for strings.

```clojure
(let [s "abcde"]
  (= (.slice s 0 -1)
     (subs s 0 (dec (count s)))))
;; => true
```

And if it's a vector, we could use `butlast` to truncate the collection:

```clojure
(= (js->clj (.slice #js [1 2 3] 0 -1))
   (butlast [1 2 3]))
```

Reply:
It's very common to do this kind of `dec` + `count` in Clojure. Unfortunately `subs` doesn't support negative indexes. But both are fine, I just tend to prefer CLJS and no interop whenever possible, even if it's a bit longer.